### PR TITLE
DDT-828: Slimming DCSA-Core part 1

### PR DESCRIPTION
**Pull request by nt-as**
This PR attempts to improve DCSA-Core by:

1. Reduce the visibility of some implementation details to reduce the number of public classes. (This scan was solely done on a class level - there are still methods could do with a reduced visibility.)  This has caused some classes to get moved as a consequence.
2. Add / Improve javadoc for some of the features (primary annotations based features were done in this run)

### CODE REVIEW

**Comment by HenrikHL** [comment]
This looks good to me - think it is a good idea for @pimg and/or @douradofrazer to give it a look also ;-)

File: src/main/java/org/dcsa/core/model/JoinedWithModel.java [hunk 1]
```diff
@@ -1,12 +1,70 @@
...
+ *
+ *     // ... an @Id field (or other fields) could go here
+ *
+ *     private String intermediateId;
+ *
+ *     @MapEntity
+ *     private InnerModel innerModel;
+ * }
+ *
+ * @Table("intermedate")
```
Comment (mp-assco) [inline src/main/java/org/dcsa/core/model/JoinedWithModel.java:48]: is the spelling intentional?

File: src/main/java/org/dcsa/core/validator/VesselIMONumberValidator.java [hunk 2]
```diff
@@ -1,10 +1,7 @@
 package org.dcsa.core.validator;
 
-import org.dcsa.core.util.ValidationUtils;
-
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
-import java.util.regex.Pattern;
 
 public class VesselIMONumberValidator implements ConstraintValidator<ValidVesselIMONumber, String> {
```
Comment (pimg) [inline src/main/java/org/dcsa/core/validator/VesselIMONumberValidator.java:6]: should we add final to this? Or do we expect variations of VesselIMONumberValidators through extending it?

**Comment by pimg** [comment]
> 

done ;)

File: src/main/java/org/dcsa/core/validator/VesselIMONumberValidator.java (hunk 2 above)
Comment (nt-as) [reply src/main/java/org/dcsa/core/validator/VesselIMONumberValidator.java]: Done

**Review by pimg — APPROVED** [review]

**Review by douradofrazer — APPROVED** [review]

**Pull request merged**